services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java

```diff
@@ -118,11 +118,14 @@ public Response updateGroup(GroupRepresentation rep) {
         this.auth.groups().requireManage(group);
 
         String groupName = rep.getName();
+
         if (ObjectUtil.isBlank(groupName)) {
             throw ErrorResponse.error("Group name is missing", Response.Status.BAD_REQUEST);
         }
 
-        checkForOrgRelatedGroupRep(session, rep);
+        if (rep.getId() != null && !group.getId().equals(rep.getId())) {
```

Comment:
Why this change in Group API is necessary?

Reply:
Because otherwise, you are accepting a rep for a group other than the one you are actually managing from the API.